src/parser/parsers/attributes.py

```diff
@@ -16,16 +16,77 @@ def __init__(self, heroes_data, localizations):
     def run(self):
         all_attributes = {}
 
+        # Extract the attributes names and group them by what category they belong to
         for hero_key, hero_value in self.heroes_data.items():
             if hero_key.startswith('hero') and hero_key != 'hero_base':
                 # Add attributes this hero contains to the master attr dict
                 all_attributes.update(self._parse_stats_ui(hero_value))
                 all_attributes.update(self._parse_shop_stat_display(hero_value))
-        # Write to
 
-        json_utils.write(
-            OUTPUT_DIR + 'json/attribute-data.json', json_utils.sort_dict(all_attributes)
-        )
+        # Determine the unlocalized name of each attribute that they should map to
+        all_attributes.update(self._map_to_unlocalized(all_attributes))
+
+        # Reorder 1st level to be in the order they are displayed in game and which happens to be the best for UX reasons
+        category_order = ['Weapon', 'Vitality', 'Spirit']
+        all_attributes = {
+            category: all_attributes[category]
+            for category in category_order
+            if category in all_attributes
+        }
+
+        # Specify order for lua as it isn't capable of iterating jsons in the order it appears
+        for category, attributes in all_attributes.items():
+            attributes_order = list(attributes)
+            all_attributes[category]["_attribute_order"] = attributes_order
+        all_attributes['_category_order'] = category_order
+
+        # Write the attributes to a json file
+        json_utils.write(OUTPUT_DIR + 'json/attribute-data.json', all_attributes)
+
+    def _map_to_unlocalized(self, all_attributes):
+        """Maps the attributes to their unlocalized names"""
+        """
+        Maps attributes as they appear in shop UI data, such as "BulletDamage" to their unlocalized names "StatDesc_BulletDamage"
+        The unlocalized name will then be localized on the front end
+        """
```

Comment:
Can these two comment blocks be combined

Reply:
done